### PR TITLE
149 implement robust dpatch attack

### PR DESCRIPTION
**Pull request by treubig26**
Initial implementation of Robust DPatch using a lightning module. This PR does not yet offer attack optimization as part of the armory-library API, but rather implements it entirely in user/example code.

Outside of editorial fixes or blatant errors, most comments on this PR will probably be addressed in a follow-on when the attack optimization is made generic and moved to be part of armory-library.

### CODE REVIEW

File: examples/src/armory/examples/object_detection/visdrone_yolov5_robustdpatch.py [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+
+        # Get targets as Tensor
+        _, _, height, width = inputs.shape
+        targets = batch.targets.get(self.target_spec)
+        yolo_targets = self._to_yolo_targets(targets, height, width, self.model.device)
+
+        # Get loss from model outputs
+        self.model.train()
+        loss_components = self.model(inputs_with_augmentations, yolo_targets)
+        loss = loss_components["loss_total"]
```
Comment (deprit) [inline examples/src/armory/examples/object_detection/visdrone_yolov5_robustdpatch.py:83]: Robust DPatch optimizes the patch by increasing the loss, equivalently minimizing the negative loss.
```
loss = - loss_components["loss_total"]
```

File: examples/src/armory/examples/object_detection/datasets/visdrone.py [hunk 2]
```diff
@@ -0,0 +1,219 @@
...
+        max_size: Maximum image size to which to resize and pad image samples
+        **kwargs: Additional keyword arguments to pass to the dataloader constructor
+
+    Return:
+        Armory object detection dataloader
+    """
+    resize = A.Compose(
+        [
+            A.LongestMaxSize(max_size=max_size),
+            A.PadIfNeeded(
```
Comment (deprit) [inline examples/src/armory/examples/object_detection/datasets/visdrone.py:35]: It may not be necessary to pad all input images to the max size, some models take a list of images as input (e.g. Faster-RCNN).

**Review by deprit — COMMENTED** [review]
The Robust DPatch example is somewhat specialized to a particular YOLOv5 model. It's not clear how much effort should be expended to generalize the code.

**Review by mwartell — APPROVED** [review]
